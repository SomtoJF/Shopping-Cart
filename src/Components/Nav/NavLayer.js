import { useEffect } from "react";
import { Link } from "react-router-dom";
import "../../styles/NavLayer.sass";

const NavLayer = () => {
	return (
		<aside id="layer">
			<ul>
				<li>
					<Link to="/">HOME</Link>
				</li>
				<li>
					<Link to="/products">PRODUCTS</Link>
				</li>
				<li>
					<Link to="/cart">CART</Link>
				</li>
			</ul>
		</aside>
	);
};
export default NavLayer;
