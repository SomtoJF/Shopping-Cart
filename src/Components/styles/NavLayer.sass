@use '../../App'
#layer
    display: flex
    flex-direction: column
    justify-content: center
    align-items: center
    position: fixed
    left: -80vw
    height: 100vh
    width: 80vw
    padding: 0px
    box-sizing: border-box
    background-color: App.$primary-color
    color: App.$secondary-color
    font-family: App.$navbar-font, system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif
    z-index: 2
    transition: left 1s

    ul
        box-sizing: border-box
        list-style-type: none
        width: 100%
        height: 30%
        display: flex
        flex-direction: column
        justify-content: center
        align-content: center
        margin: 0
        padding: 0

        li:first-child
            border-top: solid 1px white

        li
            height: 30% 
            border-bottom: solid 1px white
            display: flex
            justify-content: center
            align-items: center

            a
                display: block
                width: 20%
                // height: 100%
                color: App.$secondary-color
                text-decoration: none

        
    
