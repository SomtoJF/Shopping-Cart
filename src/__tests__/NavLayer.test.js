import React from "react";
import { render, screen } from "@testing-library/react";
// import { MemoryRouter } from "react-router-dom";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom"; // optional
import NavLayer from "../Components/Nav/NavLayer";

describe("Nav Overlay", () => {
	it("Matches Snapshot", () => {
		const { container } = render(<NavLayer />, { wrapper: MemoryRouter });
		expect(container).toMatchSnapshot();
	});
});
