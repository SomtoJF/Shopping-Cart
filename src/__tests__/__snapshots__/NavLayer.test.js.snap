// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Overlay Matches Snapshot 1`] = `
<div>
  <aside
    id="layer"
  >
    <ul>
      <li>
        <a
          href="/Shopping-Cart/"
        >
          HOME
        </a>
      </li>
      <li>
        <a
          href="/Shopping-Cart/products"
        >
          PRODUCTS
        </a>
      </li>
      <li>
        <a
          href="/Shopping-Cart/cart"
        >
          CART
        </a>
      </li>
    </ul>
  </aside>
</div>
`;
